OophionN

OophionN opened the pull request:
Добавлен скрипт выгрузки курсов валют

Maksar503 committed:
Добавить скрипт выгрузки курса валют и ReadMe

OophionN committed:
Update ReadMe.md

PhySci commented on script_currency_CB_RF/currency_CB.py:
```diff
@@ -0,0 +1,69 @@
+
+        return currency_rates
+
+    except requests.exceptions.RequestException as e:
```
try-except блог обрабатывает только ошибки выполнения запросов к серверу и не обрабатывает ошибки разбора документа.
Надо либо сделать try-except "уже", только до строчки 10, либо обрабатывать и другие ошибки.

PhySci commented on script_currency_CB_RF/currency_CB.py:
```diff
@@ -0,0 +1,69 @@
+        return currency_rates
+
+    except requests.exceptions.RequestException as e:
+        print(f"Ошибка при выполнении запроса: {e}")
```
Вместо print лучше использовать logger

PhySci commented on script_currency_CB_RF/currency_CB.py:
```diff
@@ -0,0 +1,69 @@
+
+    except requests.exceptions.RequestException as e:
+        print(f"Ошибка при выполнении запроса: {e}")
+        return None
```
Функция должна всегда возвращать один и тот же тип данных. В твоём случае - словарь. Если произошла ошибка, то возвращай пустой словарь.

PhySci commented on script_currency_CB_RF/currency_CB.py:
```diff
@@ -0,0 +1,69 @@
+        print(f"Данные сохранены в файл: {file_path}")
+
+    except IOError as e:
+        print(f"Ошибка при сохранении в файл: {e}")
```
logger

PhySci commented on script_currency_CB_RF/currency_CB.py:
```diff
@@ -0,0 +1,69 @@
+        save_to_csv(data, r"/content/sample_data/currency_rates.csv")
+
+    else:
+        print("Не удалось получить данные о курсе валют.")
```
logger

OophionN replied:
Привет, все замечания отработал, отправил на ревизию

PhySci reviewed (changes requested):
Скрипт необходимо перенести в папку src/parser вместо создания новой директории.

В той же папке, в файле utils.py есть функция setup_logging для инициализации логгера - стоит импортировать и вызвать в начале скрипта. Вот тут пример https://github.com/dataMasterskaya/DreamParser/blob/main/src/parser/app.py#L38

PhySci commented:
@OophionN необходимо обновить этот же Merge Request, а не создавать новый. Обычно для этого достаточно запушить изменения в ветку, которую ты хочешь влить. Сейчас вижу два новых MR и два новых файла src/parser/parser_CB_ver3.py и src/parser/parser_CB_ver2.py. Так лучше не делать.

- setup_logging лучше всё таки импортировать из utils, иначе возникает дублирование кода.
  
- сохранять логи и результаты работы необходимо по относительным путям, а не абсолютным. Иначе это работает только у тебя. И лучше использовать linux пути, т.к. парсер будет работать из linux контейнера.

OophionN committed:
update parser

Добавить вызов utils скорректировать относительные пути для сохранения результатов

OophionN commented:
@PhySci скорректировал код, сделал вызов utils и поправил пути. С ветками сори, только неделю осваиваю этот инструмент. выгрузил в ветку OophionN в директорию script_currency_CB_RF

OophionN committed:
Delete currency_CB.py

Удалить в связи с загрузкой обновленной версии - актуальный файл parser_CB_ver3

OophionN commented:
@PhySci Я удалил ветки которые насоздавал OophionN-patch-1 и OophionN-patch-2, последняя версия как указал выше в ветке OophionN.
